Error when create modules with spaces on name

**Issue description:**
If you create a module with spaces on name using the Module UI, the editor cannot create module class correctly.

![image](https://github.com/FlaxEngine/FlaxEngine/assets/79365912/fdf17a53-6082-4dd2-8089-43f7a94e33ad)
![image](https://github.com/FlaxEngine/FlaxEngine/assets/79365912/5352c85b-0b65-4fd6-87a6-42682bc7d072)


```
[ 00:09:29.000 ]: [Warning] Command exited with error code=1
[ 00:09:29.000 ]: [Error] Failed to run build tool, result: 1
[ 00:09:29.999 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(4,21): error CS1514: { expected
[ 00:09:29.999 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(4,21): error CS1513: } expected
[ 00:09:29.999 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(4,21): error CS8803: Top-level statements must precede namespace and type declarations.
[ 00:09:29.999 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(4,38): error CS1002: ; expected
[ 00:09:29.999 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(5,2): error CS1513: } expected
[ 00:09:29.999 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(27,1): error CS1022: Type or namespace definition, or end-of-file expected
[ 00:09:29.999 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(7,26): error CS0116: A namespace cannot directly contain members such as fields, methods or statements
[ 00:09:30.011 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(7,26): error CS0115: '<invalid-global-code>.Init()': no suitable method found to override
[ 00:09:30.013 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(16,26): error CS0115: '<invalid-global-code>.Setup(BuildOptions)': no suitable method found to override
[ 00:09:30.013 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(4,21): error CS8805: Program using top-level statements must be an executable.
[ 00:09:30.014 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(4,28): error CS0119: 'GameModule' is a type, which is not valid in the given context
[ 00:09:30.014 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(4,21): warning CS0164: This label has not been referenced
[ 00:09:30.014 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(9,14): error CS0117: 'object' does not contain a definition for 'Init'
[ 00:09:30.014 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(12,9): error CS0103: The name 'BuildNativeCode' does not exist in the current context
[ 00:09:30.014 ]: [Info] /home/capy/Documents/FlaxSamples-master/BasicTemplate/Source/Module Test/Module Test.Build.cs(18,14): error CS0117: 'object' does not contain a definition for 'Setup'
[ 00:09:30.014 ]: [Info] Exception: Failed to build assembly.
[ 00:09:30.014 ]: [Info] Stack trace:
[ 00:09:30.018 ]: [Info]    at Flax.Build.Assembler.Build() in /home/capy/Documents/Dev/FlaxEngine/Source/Tools/Flax.Build/Build/Assembler.cs:line 179
[ 00:09:30.018 ]: [Info]    at Flax.Build.Builder.GenerateRulesAssembly() in /home/capy/Documents/Dev/FlaxEngine/Source/Tools/Flax.Build/Build/Builder.Rules.cs:line 193
[ 00:09:30.018 ]: [Info]    at Flax.Build.Program.Main() in /home/capy/Documents/Dev/FlaxEngine/Source/Tools/Flax.Build/Program.cs:line 116
[ 00:09:30.018 ]: [Info] Total time: 00:00:02.1263282
[ 00:09:30.027 ]: [Warning] Command exited with error code=1
[ 00:09:30.027 ]: [Error] Failed to run build tool, result: 1

```

**Flax version:**
1.7 master

--- a/Source/Editor/Windows/ContentWindow.ContextMenu.cs
+++ b/Source/Editor/Windows/ContentWindow.ContextMenu.cs
@@ -369,7 +369,7 @@ namespace FlaxEditor.Windows
                 }
 
                 var pluginPath = Path.Combine(Globals.ProjectFolder, "Source", nameTextBox.Text);
-                if (Directory.Exists(pluginPath))
+                if (!IsValidModuleName(nameTextBox.Text) || Directory.Exists(pluginPath))
                 {
                     nameTextBox.BorderColor = Color.Red;
                     nameTextBox.BorderSelectedColor = Color.Red;
@@ -429,6 +429,12 @@ namespace FlaxEditor.Windows
             submitButton.Clicked += () =>
             {
                 // TODO: Check all modules in project including plugins
+                if (!IsValidModuleName(nameTextBox.Text))
+                {
+                    Editor.LogWarning("Invalid module name. Module names cannot contain spaces, start with a number or contain non-alphanumeric characters.");
+                    return;
+                }
+                
                 if (Directory.Exists(Path.Combine(Globals.ProjectFolder, "Source", nameTextBox.Text)))
                 {
                     Editor.LogWarning("Cannot create module due to name conflict.");
@@ -459,6 +465,17 @@ namespace FlaxEditor.Windows
                 popup.Hide();
                 button.ParentContextMenu.Hide();
             };
+        }
+
+        private static bool IsValidModuleName(string text)
+        {
+            if (text.Contains(' '))
+                return false;
+            if (char.IsDigit(text[0]))
+                return false;
+            if (text.Any(c => !char.IsLetterOrDigit(c) && c != '_'))
+                return false;
+            return true;
         }
     }
 }

